state/upgrades_test.go

```diff
@@ -372,25 +372,25 @@ func (s *upgradesSuite) TestEnsureInitalRefCountForExternalSecretBackends(c *gc.
 	s.assertUpgradedData(c, EnsureInitalRefCountForExternalSecretBackends, expectedData)
 }
 
-func (s *upgradesSuite) TestEnsureApplicationCharmOriginsHaveRevisions(c *gc.C) {
+func (s *upgradesSuite) TestEnsureApplicationCharmOriginsNormalise(c *gc.C) {
 	ch := AddTestingCharm(c, s.state, "dummy")
 	platform := Platform{OS: "ubuntu", Channel: "20.04"}
+	rev := 6
 	_ = addTestingApplication(c, addTestingApplicationParams{
 		st:   s.state,
 		name: "my-local-app",
 		ch:   ch,
 		origin: &CharmOrigin{
 			Source:   charm.Local.String(),
```

Comment:
This is bad - sorry I didn't catch it earlier.  charm.Local == corecharm.Local - however charm.CharmHub != corecharm.CharmHub. Always use a corecharm Source in the charm origin.

Reply:
Yes good spot. That is something I have only learnt recently